search/search-client-elasticsearch/src/main/java/io/camunda/search/es/clients/ElasticsearchSearchClient.java

```diff
@@ -45,14 +90,93 @@ public <T> Either<Exception, SearchQueryResponse<T>> search(
       final SearchResponseTransformer<T> searchResponseTransformer = getSearchResponseTransformer();
       final SearchQueryResponse<T> response = searchResponseTransformer.apply(rawSearchResponse);
       return Either.right(response);
-    } catch (final IOException ioe) {
+    } catch (final IOException | ElasticsearchException ioe) {
       return Either.left(ioe);
```

Comment:
Can we throw `SearchQueryExecutionException` from here to:
- lighten the return type of this method from  `Either<Exception, SearchQueryResponse<T>>` to `SearchQueryResponse<T>`
- avoid the service layer handing this exception, the same code `fold(e->{throw new SearchQueryExecutionException("Failed to execute search query", e);})` is applied to all service classes